src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1830,6 +1831,28 @@ private int getStringLiteral(CORINFO_MODULE_STRUCT_* module, uint metaTOK, char*
             return str.Length;
         }
 
+        private nuint printObjectDescription(void* handle, byte* buffer, nuint bufferSize)
+        {
+            Debug.Assert(bufferSize > 0 && handle != null && buffer != null);
+
+            int bufferSize32 = checked((int)bufferSize);
+
+            ReadOnlySpan<char> objStr = HandleToObject(handle).ToString();
+            var bufferSpan = new Span<byte>(buffer, bufferSize32);
+            Utf8.FromUtf16(objStr, bufferSpan, out _, out int written);
+            if (written >= bufferSize32)
+            {
+                // Trim data and null-terminate it
+                bufferSpan[bufferSize32 - 1] = 0;
+            }
+            else
+            {
+                bufferSpan[written] = 0;
+            }
+
+            return (nuint)objStr.Length;
```

Comment:
This does not match the documented contract. This is the UTF-16 length. The contract says that it should be the full UTF-8 length.

Reply:
Oops, indeed, fixed